Name: Add Priority Discard Btns
Version: 1.2.1-12120
Author: mamouton
Description: This will place a high priority icon and a discard message icon at the bottom of your current message. Special thanks to Tibfib for adding the confirmation for the discard message and removing the options from the menu area.

--- .orig/usr/palm/applications/com.palm.app.email/app/controllers/compose-assistant.js
+++ /usr/palm/applications/com.palm.app.email/app/controllers/compose-assistant.js
@@ -79,6 +79,7 @@
 				visible: true,
 				menuClass: 'palm-white',
 				items: [
+					{label:$L('Priority'), icon:'priority', command:'priority'},
 					{label:$L('Attach'), icon:'attach', command:'attach'},
 					{label:$L('Send'), icon:'send', command:'send'}
 				]};
@@ -80,7 +81,8 @@
 				menuClass: 'palm-white',
 				items: [
 					{label:$L('Attach'), icon:'attach', command:'attach'},
-					{label:$L('Send'), icon:'send', command:'send'}
+					{label:('Send'), icon:'send', command:'send'},
+					{label:$L('Delete'),icon:'delete', command:'cancel'}
 				]};
 
 		this.draftIsDirty = false;
@@ -89,8 +91,8 @@
 		this.appMenuModel = {
 				visible:true,
 				items: [
-					this.priorityFlagMenuItem,
-					{label:$L('Discard Message'), shortcut:'d', command:'cancel'},
+					//this.priorityFlagMenuItem,
+					//{label:$L('Discard Message'), shortcut:'d', command:'cancel'},
					{label:$L('Save as Draft'), shortcut:'s', command:'save', disabledProperty:true }
 				]};
 		this.controller.setupWidget(Mojo.Menu.appMenu, {richTextEditMenu: true}, this.appMenuModel);
@@ -185,7 +187,8 @@
 		}
 
 		// Delayed a little, since we want the header part of render as quickly as possible
-		ComposeAssistant.onLoad.defer(this.controller, this.email);
+ComposeAssistant.onLoad.defer(this.controller, this.email);
+this.controller.window.PalmSystem.setWindowOrientation("free");
 	},
 
 	cleanup: function() {
@@ -236,7 +239,15 @@
 					break;
 
 				case 'cancel':
-					this.cancelCompose();
+					this.controller.showAlertDialog({
+					onChoose: function(value) {if (value == 'yes') {this.cancelCompose();}},
+					title: $L("Discard this Draft?"),
+					message: $L("This cannot be undone"),
+					choices:[
+						{label:$L('Yes'), value:"yes", type:'negative'},  
+						{label:$L("No"), value:"no", type: 'dismiss'}
+							]
+					});
 					break;
 						
 				case 'priority':
